toolchain/parse/handle_paren_expr.cpp

```diff
@@ -6,6 +6,38 @@
 
 namespace Carbon::Parse {
 
+auto HandleOnlyParenExpr(Context& context) -> void {
+  auto state = context.PopState();
+
+  // Advance past the open paren.
+  context.AddLeafNode(NodeKind::ExprOpenParen,
+                      context.ConsumeChecked(Lex::TokenKind::OpenParen));
+
+  context.PushState(state, State::ParenExprFinish);
```

Comment:
WDYT of moving the shared logic out to a helper function instead of pushing a state? I was looking at the SkipTo in OnlyParenExprFinish and it took me a moment to figure out where the position was being advanced past the close paren, which might've been more apparent with a helper function. Since there isn't intermediate processing, I don't think the state stack is strictly required.

Reply:
The helper function is only one line, which makes it a bit suspect. But making that change ended up with a better tree, I think, so I kept the change.